packages/extension-driver-duckdb/src/lib/duckdbDataSource.ts

```diff
@@ -198,9 +197,28 @@ export class DuckDBDataSource extends DataSource<any, DuckDBOptions> {
         (err: any) => {
           if (err) reject(err);
           this.logger.debug(`Table created, name = ${tableName}`);
+          // create indexes if set the options
+          if (indexes)
+            this.createIndexes(connection, schema, tableName, indexes);
           resolve();
         }
       );
     });
   }
+
+  // create indexes for table
+  private createIndexes(
+    connection: duckdb.Connection,
+    schema: string,
+    tableName: string,
+    indexes: Record<string, string>
+  ) {
+    Object.entries(indexes).forEach(([index, column]) => {
+      // duckdb could not create duplicate index, so here should drop index if exists and create again.
+      connection.run(`DROP INDEX IF EXISTS ${index} ON ${schema}.${tableName}`);
```

Comment:
Could you tell me the purpose of recreating an index with the same name on the same table?

Reply:
The reason is that we will refresh and call the importing method again, but the duckdb could create duplicate index names.

Although we could query whether the index from duckdb and check it exists. But it may do more steps:
1. query the table
2. get the result and do check if exist
3. If not exist, create the index

So, for the simple way, we could drop and recreate directly.

PS: I discovered my `DROP INDEX IF EXISTS ${index} ON ${schema}.${tableName}` syntax is inccoret, the duckdb couldn't drop the index on the specific table. https://duckdb.org/docs/sql/indexes#drop-index

Reply:
Thanks for @onlyjackfrost feedback the worried cost a lot of time when dropping the index if the table has thousands of thousands ( seems it's an issue in OLTP database, but not sure for OLAP database), we could arrange the performance testing in the future and if it's a issue, we will change back to the above checking steps.